### ISSUE TITLE
Unify capitalization of plugin names / provide guidelines for plugin naming

### ISSUE THREAD

**Issue by goanpeca** [napari/napari#3102]
## 🚀 Feature

<img width="198" alt="Screen Shot 2021-07-29 at 9 54 28 AM" src="https://user-images.githubusercontent.com/3627835/127514797-65cf3a30-2a5a-4008-99dd-cfee8fe8491f.png">

Some start with lowercase some other do not, and plugins should provide a display language so that capitalization rules will change for different languages (if they apply)

## Motivation

Provide a consiste UI

## Pitch

The mix of lowercase and uppercase on plugin/hooks names makes the UI inconsistent.
Also it would be great to provide naming guideling for users and developers.

---

Pinging @liaprins-czi

**Comment by liaprins-czi** [comment]
> The mix of lowercase and uppercase on plugin/hooks names makes the UI inconsistent.
> Also it would be great to provide naming guideling for users and developers.

Yes I think we will be doing this as part of the hub's effort to have better plugin names shown. (@neuromusic for reference)

@goanpeca Do verify, are you referring to the actual name of the plugin, or to commands that the plugin includes and that subsequently appear in menubar when using it? (or both?)

**Comment by goanpeca** [comment]
> Do verify, are you referring to the actual name of the plugin, or to commands that the plugin includes and that subsequently appear in menubar when using it? (or both?)

I meant the name of the plugin that will appear on the plugins menu, if they provide a dockwidget hookspec.

**Comment by liaprins-czi** [comment]
Okay great!

**Comment by haesleinhuepf** [comment]
Hi all,

great to see this discussion here! I just renamed some of my plugins so that their name highlights more "What" the plugin is doing. I would recommend that to all plugin developers in general. We could then mid/long term have a menu that contains
* Filtering
* Segmentation
* Measurements

Instead of technical terms such as
* clEsperanto
* StarDist
* Regionprops

From a users perspective, that might be very welcome.

Furthermore, I would like to add a technical aspect: Menu order in the plugins menu. Would it be possible to sort it alphabetically? At the moment it appears a bit random and a bit counter-alphabetically.
![image](https://user-images.githubusercontent.com/12660498/127767315-78880dea-d52b-4be1-b977-56fcd1314e4e.png)

Cheers,
Robert

**Comment by tlambert03** [comment]
oof @haesleinhuepf 😝 [this bit](https://github.com/clEsperanto/napari_pyclesperanto_assistant/blob/8667d44526729582205ce02444706f4ed11b22c3/setup.py#L56-L66) right here is definitely not what was intended by the entry_point mechanism.  I'd point you to #3035 for most of the conversation on this topic (I even proposed recently that we [drop support](https://github.com/napari/napari/issues/3035#issuecomment-882752676) for multiple entry_points per package).  An entry point is just that: a place for napari to "enter" your package to look for things.  It was never intended as a menu structure mechanism.  

> I would recommend that to all plugin developers in general.

please don't? 😂 

I recognize that we don't yet have an infrastructure that supports complex menu nesting... and I know you're itching to start exposing features... so I feel like the best we can probably suggest at this point is for you to create your _own_ **single** QMenu that is dropped into the plugins menu.  (this too isn't "supported", and will likely break eventually... but we're just not "there" yet)

**Comment by haesleinhuepf** [comment]
> please don't? 😂

Well, the headline of this issue is "provide guidelines for plugin naming". And I think plugins should have meaningful names such as "AlgorithmX-Segmentation" or "Measurement: MethodY". Not sure why you oppose here.

At the moment, users have a hard time because the menu is full of "abcde-napari: abcde", "ThingStar" and "CellPlugin". In many cases it's not obvious from the menu entry name what the plugin does. That's why I also attempted to rename the entries of "clEsperanto" as it also doesn't tell much about what it's doing. I'm just trying to make it more accessible for users.

**Comment by tlambert03** [comment]
> And I think plugins should have meaningful names such as "AlgorithmX-Segmentation" or "Measurement: MethodY". Not sure why you oppose here.

I oppose because `Measurement: MethodY` is not a plugin ... it's a command.  And while I get that "provide guidelines for plugin naming" is in the title of this post...  I think we mean more along the lines of https://github.com/napari/napari/issues/3035, (where the general desire was to go the other direction with entry points).

> At the moment, users have a hard time because the menu is full of "abcde-napari: abcde", "ThingStar" and "CellPlugin". In many cases it's not obvious from the menu entry name what the plugin does. 

Again, we completely and whole-heartedly agree here.  But the solution isn't to try to figure out what we can do with the system _as is_ (for instance, by adding a lot of entry points in a single package), but rather to step back for a moment and decide what the "best" way to do this is, and make _that_ a reality.  I know you're trying to make it more accessible for users... but you have to also recognize that plugins will just start mimicking other plugins.  and users will be ever _more_ confused when the whole question of "what is a plugin in napari anyway?  is it a package?  is it a menu item"

We're routinely having conversations about how to expose menu items in a nestable and organized way.  Specifically with your sort of use case in mind... So, when I said "please don't", I meant please don't encourage people to make a handful of plugins in a single package, each with names like "1: segmentation", etc...  Really, that should be a single plugin, and we ask everyone to be patient while we build a more stable foundation for inner-plugin organization

**Comment by psobolewskiPhD** [comment]
I mentioned this in the image.sc thread, but coming from a neophyte PoV, I strongly feel the Plugins menu should remain sane and well organized, 1 item per plugin. What would be nice though is to be able to activate more robust plugins to enable have their own dedicated menu. Then it would be easy to access those features and know clearly what plugin they belong to. I prefer a menu to a panel in napari because of limited screen real estate and neigh unlimited menu bar space.

**Comment by tlambert03** [comment]
>  I strongly feel the Plugins menu should remain sane and well organized, 1 item per plugin.  What would be nice though is to be able to activate more robust plugins to enable have their own dedicated menu.

This is absolutely the goal.  work actively in progress.  We'll be putting out some more public proposals in the not too distant future.

**Comment by liaprins-czi** [comment]
> > Do verify, are you referring to the actual name of the plugin, or to commands that the plugin includes and that subsequently appear in menubar when using it? (or both?)

> I meant the name of the plugin that will appear on the plugins menu, if they provide a dockwidget hookspec.

@neuromusic Is this something that is within the scope of the upcoming hub work for "Simplified plugin names"?

**Comment by neuromusic** [comment]
@liaprins-czi yes, I think so. in a convo w/ the Plugin Working Group (@nclack @tlambert03 & @Carreau), it was proposed that this would be a unicode field in the to-be-developed manifest file... then, either the napari hub or viewer could render it in the appropriate places
